src/block/Magma.php

```diff
@@ -27,23 +27,19 @@
 use pocketmine\entity\Living;
 use pocketmine\event\entity\EntityDamageByBlockEvent;
 use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\math\Facing;
 
 class Magma extends Opaque{
 
 	public function getLightLevel() : int{
 		return 3;
 	}
 
-	public function hasEntityCollision() : bool{
-		return true;
-	}
-
-	public function onEntityInside(Entity $entity) : bool{
-		if($entity instanceof Living && !$entity->isSneaking()){
+	public function onEntityCollide(Entity $entity, int $face) : void{
+		if($face === Facing::DOWN && $entity instanceof Living && !$entity->isSneaking()){
```

Comment:
Isn't that trivial? If the entity is above the block, we get damage and the face is below. Have I misunderstood something?

Reply:
No, that's good.
I add the condition ``$face === Facing::DOIWN`` after suggesting @IvanCraft623 to allow detection for each face and not just downwards.
Without this verification, magma blocks cause damage on all faces and not only on the down face

Reply:
Perhaps we've misunderstood each other, but what I mean is that the $face parameter located in the block class must be the face of the collided block and not the "face" of the collided entity (which is what I'm supposed to be now).

Reply:
Indead, i fix this